### PR TITLE
Add size to Save function

### LINKED ISSUE #15253: git LFS and storage type minio: Massive memory usage
- Gitea version (or commit ref): 1.14.0-rc2
- Git version: 2.30.2
- Operating system: docker
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Log gist (heap, allocs, gitea.log): 
[Logs.zip](https://github.com/go-gitea/gitea/files/6248792/Logs.zip)

## Description
This is a follow up issue from the discord general channel. 
Running git lfs push with very small LFS files (330kb in size total) and storage type minio causes a massive use of memory of gitea (around 2.2 GB). Pushing another round of small files quickly exceeds 5GB.

I’ve attached a zip containing a docker compose and test data so that you can reproduce the issue easily.
[docker.zip](https://github.com/go-gitea/gitea/files/6248812/docker.zip)

Reproduction steps:
1. Extract docker.zip
2. Adjust IP address for server.root_url and storage.minio.minio_endpoint in config/app.ini
3. Run: docker-compose -f docker-compose.yml up -d
4. Go to localhost and setup gitea
5. Create a new repo + initialize
6. Run: git clone 
7. Cd into cloned repo
8. Run: git lfs install
9. Run: git lfs track *.png
10. Copy provided PNG images to repository
11. Run: git add .
12. Run: git lfs status     - and verify that LFS files are tracked correctly
13. Run: git commit and push
14. Observe gitea memory usage spikes from 180 mb to around 2.2 GB

### PR DESCRIPTION
**Pull request by zeripath**
This PR proposes an alternative solution to #15255 - just add the size to the
save function. Yes it is less apparently clean but it may be more correct.

Close #15255
Fix #15253

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by lunny** [comment]
We need some comment to say the size could be `-1`.

**Review by 6543 — APPROVED** [review]

File: modules/storage/storage.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ type Object interface {
 // ObjectStorage represents an object storage to handle a bucket and files
 type ObjectStorage interface {
 	Open(path string) (Object, error)
-	Save(path string, r io.Reader) (int64, error)
+	Save(path string, r io.Reader, size int64) (int64, error)
```
Comment (6543) [inline modules/storage/storage.go:69]:
```suggestion
	// Save store a object, if size is unknown set -1
	Save(path string, r io.Reader, size int64) (int64, error)
```

**Review by lunny — APPROVED** [review]

**Comment by 6543** [comment]
:rocket:

**Comment by lunny** [comment]
Maybe we can wait for adding PartSize to `PutObjectOptions` before send back port.

**Comment by zeripath** [comment]
@lunny I don't understand - do you have a PR for that?

**Comment by lunny** [comment]
When `-1` is given, minio sdk will spend max 5G memory if partSize given `0`. So I think we could also give a fixed partSize even size is `-1`.

**Pull request merged**